sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamAggregationRel.java

```diff
@@ -113,10 +117,59 @@ public RelWriter explainTerms(RelWriter pw) {
 
   @Override
   public PTransform<PCollectionList<Row>, PCollection<Row>> buildPTransform() {
-    return new Transform();
+    Schema inputSchema = CalciteUtils.toSchema(getInput().getRowType());
+    Schema outputSchema = CalciteUtils.toSchema(getRowType());
+    List<AggregationCombineFnAdapter> aggregationAdapters =
+        getNamedAggCalls()
+            .stream()
+            .map(aggCall -> AggregationCombineFnAdapter.of(aggCall, inputSchema))
+            .collect(toList());
+
+    return new Transform(
+        windowFn, windowFieldIndex, inputSchema, getGroupSet(), aggregationAdapters, outputSchema);
   }
 
-  private class Transform extends PTransform<PCollectionList<Row>, PCollection<Row>> {
+  private static class Transform extends PTransform<PCollectionList<Row>, PCollection<Row>> {
```

Comment:
I think making this class static is not necessary even if you want to pass a bunch of parameters to Transform. Seems like the main benefit of static inner class is not requiring creating an outer class's instance, which does not apply here.

https://stackoverflow.com/a/22098846/10055573

Reply:
It is not necessary but it helps, e.g. if you want to embed lambdas then it's easy to have something non-serializable or something that you don't actually want to serialize in the closure. Errors that we throw in these cases are non-informational, making it static and keeping only things that are needed by the class in the class itself makes it easier to trace everything